hack/generate-operator-bundle.py

```diff
@@ -4,115 +4,199 @@
-        {
-            'rules': hive_role['rules'],
-            'serviceAccountName': 'default',
-        })
-
-# Add our hiveadmission role to the CSV:
-with open('config/hiveadmission/hiveadmission_rbac_role.yaml', 'r') as stream:
-    hiveadmission_role = yaml.load(stream, Loader=yaml.SafeLoader)
-    csv['spec']['install']['spec']['clusterPermissions'].append(
-        {
-            'rules': hiveadmission_role['rules'],
-            'serviceAccountName': 'hiveadmission',
-        })
-
-# Add our hiveapi role to the CSV:
-with open('config/apiserver/hiveapi_rbac_role.yaml', 'r') as stream:
-    hiveapi_role = yaml.load(stream, Loader=yaml.SafeLoader)
-    csv['spec']['install']['spec']['clusterPermissions'].append(
-        {
-            'rules': hiveapi_role['rules'],
-            'serviceAccountName': 'hiveapi-sa',
-        })
-
-# Add our deployment spec for the hive operator:
-with open('config/operator/operator_deployment.yaml', 'r') as stream:
-    operator_components = []
-    operator = yaml.load_all(stream, Loader=yaml.SafeLoader)
-    for doc in operator:
-        operator_components.append(doc)
-    operator_deployment = operator_components[1]
-    csv['spec']['install']['spec']['deployments'][0]['spec'] = operator_deployment['spec']
-
-# Update the deployment to use the defined image:
-csv['spec']['install']['spec']['deployments'][0]['spec']['template']['spec']['containers'][0]['image'] = hive_image
-
-# Update the versions to include git hash:
-csv['metadata']['name'] = "hive-operator.v%s" % full_version
-csv['spec']['version'] = full_version
-csv['spec']['replaces'] = "hive-operator.v%s" % prev_version
-
-# Set the CSV createdAt annotation:
-now = datetime.datetime.now()
-csv['metadata']['annotations']['createdAt'] = now.strftime("%Y-%m-%dT%H:%M:%SZ")
-
-# Write the CSV to disk:
-csv_filename = "hive-operator.v%s.clusterserviceversion.yaml" % full_version
-csv_file = os.path.join(version_dir, csv_filename)
-with open(csv_file, 'w') as outfile:
-    yaml.dump(csv, outfile, default_flow_style=False)
-print("Wrote ClusterServiceVersion: %s" % csv_file)
 
+logging.basicConfig()
+logger = logging.getLogger(__name__)
+logger.setLevel(logging.DEBUG)
+
+
+OSD_HIVE_IMAGE_DEFAULT = 'quay.io/app-sre/hive'
+OPERATORHUB_HIVE_IMAGE_DEFAULT = 'quay.io/openshift-hive/hive'
+
+
+def current_operatorhub_version(args=None):
+    result = subprocess.check_output("git describe --abbrev=0", shell=True).strip()
+    # remove the v -- OLM bundles need to start with the digit
+    result = result[1:]
+    return result
+
+
+def generate_operatorhub_version(args=None, prev_version=current_operatorhub_version()):
+    # bump the zstream
+    previous_z = prev_version[-1]
+    previous_z_int = int(previous_z)
+    new_z_int = previous_z_int + 1
+    new_version = "%s%s" % (prev_version[:-1], new_z_int)
+    return new_version
+
+
+def generate_osd_version(args=None):
+    result = subprocess.check_output("git describe", shell=True).strip()
```

Comment:
Damn, that's handy. Was not aware of this command.

Will this roll out safely as an upgrade? Considering they're on 0.1.x now it must be ok as we're going to 1.x.x.